src/librustc/hir/lowering.rs

```diff
@@ -3284,7 +3295,7 @@ impl<'a> LoweringContext<'a> {
             // This is the normal case.
             AnonymousLifetimeMode::PassThrough => self.new_implicit_lifetime(span),
 
-            AnonymousLifetimeMode::ReportError => self.new_error_lifetime(None, span),
+            AnonymousLifetimeMode::ReportError => self.new_missing_lifetime(None, span),
```

Comment:
This is always called with `None`, so you might as well just use `new_implicit_lifetime` here, no?

Reply:
Well, shame of me. Thanks for pointing out XD